### DISCUSSION TITLE
[BUG?] `navigation.activeVisualElement` in `PartialBentoConfig` does not gain the partial behaviour

### DISCUSSION THREAD

**Discussion by mperrando** [buildo/bento-design-system#808] [Q&A] [answered]
The user must define all the object properties and not only the ones they would override.

Reporo step:
create a partial config with the empty element
```ts
const badConfig: PartialBentoConfig = {
  navigation: {
    activeVisualElement: {},
  },
};
```

**Comment by gabro** [comment]
Thanks for reporting; it is indeed a bug in the types (the runtime should work ok).
I've tracked it down to our `PartialDeep` implementation. I'm working on a fix!

**Reply by gabro** [reply, marked as answer]
Fixed in https://github.com/buildo/bento-design-system/pull/809
